Doc/library/io.rst

```diff
@@ -405,18 +418,17 @@ I/O Base Classes
 
    .. method:: seek(offset, whence=SEEK_SET, /)
 
-      Change the stream position to the given byte *offset*.  *offset* is
-      interpreted relative to the position indicated by *whence*.  The default
-      value for *whence* is :data:`SEEK_SET`.  Values for *whence* are:
+      Change the stream position to the given byte *offset*,
+      and return the new absolute position as an integer.
+      *offset* is interpreted relative to the position indicated by *whence*.
+      Valid values for *whence* are:
 
-      * :data:`SEEK_SET` or ``0`` -- start of the stream (the default);
+      * :data:`SEEK_SET` -- seek from the start of the stream (the default);
         *offset* should be zero or positive
-      * :data:`SEEK_CUR` or ``1`` -- current stream position; *offset* may
-        be negative
-      * :data:`SEEK_END` or ``2`` -- end of the stream; *offset* is usually
-        negative
-
-      Return the new absolute position.
+      * :data:`SEEK_CUR` -- seek from current stream position;
+        *offset* may be negative
+      * :data:`SEEK_END` -- seek from the end of the stream;
+        *offset* is usually negative
 
```

Comment:
Given the pitfalls of seeking with files in text-mode, perhaps ``IOBase.seek`` should link to ``TextIOBase.seek``, which has more specific information on the limitations here:

```suggestion

      .. note::
         When working with a text stream :py:meth:`!seek` has limitations
         beyond the general interface described here,
         which are described in more detail in :py:meth:`TextIOBase.seek`.
```